### PR TITLE
Add directoryConfig to globalData

### PR DESCRIPTION
**Pull request by hbalenda**
The directory config is not available to immediately invoked plugins (like vite), so previously we were passing directories explicitly. Additionally, the `publicDir` was not available in the global data at all, so the `iiifConfig` was previously parsing the vite options to get it.

Changes:
- Provide `inputDir` `outputDir` and `publicDir` to globalData plugin and add them to globalData under `directoryConfig` key
- Decouple `iiifConfig` from vite plugin

### CODE REVIEW

File: packages/11ty/.eleventy.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ const { error } = chalkFactory('eleventy config')
 
 const inputDir = process.env.ELEVENTY_INPUT || 'content'
 const outputDir = process.env.ELEVENTY_OUTPUT || '_site'
-const publicDir = 'public'
+const publicDir = process.env.ELEVENTY_ENV === 'production' ? 'public' : false // yes, false
```
Comment (mphstudios) [inline packages/11ty/.eleventy.js:42]: What does `// yes, false` mean?

**Review by mphstudios — COMMENTED** [review]
@hbalenda it looks like `eleventyConfig.eleventyFiles` is available to the plugins though does not have a `publicDir` property?

File: packages/11ty/.eleventy.js (hunk 1 above)
Comment (hbalenda) [inline packages/11ty/.eleventy.js:42]:
```suggestion
const publicDir = process.env.ELEVENTY_ENV === 'production' ? 'public' : false // publicDir should be set explicitly to false in development 
```

**Review by mphstudios — APPROVED** [review]

**Comment by hbalenda** [comment]
@mphstudios that property is not available to the plugins that we invoke without using `addPlugin` (for example, the vite plugin)

**Pull request merged**